File: Examples/BankAccount/BankAccount.java
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.1

public final class BankAccount {
    private BankAccount() {}

    private static final class Account {
        private final String owner;
        private long balance;

        Account(String owner, long openingBalance) {
            this.owner = owner;
            this.balance = openingBalance;
        }

        void deposit(long amount) {
            balance += amount;
        }

        boolean withdraw(long amount) {
            if (amount > balance) {
                return false;
            }
            balance -= amount;
            return true;
        }

        void print() {
            System.out.printf("%s: %d%n", owner, balance);
        }
    }

    public static void main(String[] args) {
        var account = new Account("Ada", 250);
        account.deposit(75);
        account.withdraw(40);
        account.print();
    }
}

File: Examples/BankAccount/BankAccount.cs
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.1

using System;

namespace Examples.BankAccount;

public sealed class Account(string owner, decimal openingBalance)
{
    public string Owner { get; } = owner;
    public decimal Balance { get; private set; } = openingBalance;

    public void Deposit(decimal amount) => Balance += amount;

    public bool Withdraw(decimal amount)
    {
        if (amount > Balance)
        {
            return false;
        }
        Balance -= amount;
        return true;
    }
}

public static class BankAccount
{
    public static void Main()
    {
        var account = new Account("Ada", 250m);
        account.Deposit(75m);
        account.Withdraw(40m);
        Console.WriteLine($"{account.Owner}: {account.Balance}");
    }
}
